arangod/Aql/QueryContext.h

```diff
@@ -103,6 +103,10 @@ class QueryContext {
 
   aql::Ast* ast();
 
+  /// @brief Acquire a lock_guard on the mutex to serialize concurrent snippet
+  /// execution
+  std::lock_guard<std::mutex> acquireLockGuard();
```

Comment:
NIT

if you move this to header can be written without mutex type, auto in return type, template deduction for return statement